### ISSUE TITLE
synchronous_mode

### ISSUE THREAD

**Issue by jamessewell** [zalando/patroni#390]
Hello,

My understanding of `synchronous_mode `was that Patroni forces a single node into `synchronous_standby_names` to ensure that the current sync standby is always known - but this doesn't seem to be happening.

Am I misinterpreting how this works? Details below:

```
{
  "retry_timeout": 10,
  "postgresql": {
    "use_slots": false,
    "parameters": {
      "synchronous_standby_names": "*"
    },
    "use_pg_rewind": true
  },
  "synchronous_mode": true,
  "maximum_lag_on_failover": 1048576,
  "loop_wait": 10,
  "ttl": 30
}
```

```
{
  "database_system_identifier": "6382399396547239070",
  "postmaster_start_time": "2017-02-02 17:49:22.839 AEDT",
  "xlog": {
    "location": 100663840
  },
  "patroni": {
    "scope": "postgresql",
    "version": "1.1"
  },
  "replication": [
    {
      "sync_state": "sync",
      "sync_priority": 1,
      "client_addr": "192.168.2.247",
      "state": "streaming",
      "application_name": "postgresql-192.168.2.247",
      "usename": "replication"
    },
    {
      "sync_state": "potential",
      "sync_priority": 1,
      "client_addr": "192.168.2.245",
      "state": "streaming",
      "application_name": "postgresql-192.168.2.245",
      "usename": "replication"
    },
    {
      "sync_state": "potential",
      "sync_priority": 1,
      "client_addr": "192.168.2.246",
      "state": "streaming",
      "application_name": "postgresql-192.168.2.246",
      "usename": "replication"
    },
    {
      "sync_state": "potential",
      "sync_priority": 1,
      "client_addr": "192.168.2.248",
      "state": "streaming",
      "application_name": "postgresql-192.168.2.248",
      "usename": "replication"
    }
  ],
  "state": "running",
  "role": "master",
  "server_version": 90505
}

```

```
 -bash-4.2$ psql
psql (9.5.5)
Type "help" for help.

postgres=# select * from pg_stat_replication;
 pid  | usesysid |   usename   |     application_name     |  client_addr  | client_hostname | client_port |         backend_start         |
 backend_xmin |   state   | sent_location | write_location | flush_location | replay_location | sync_priority | sync_state
------+----------+-------------+--------------------------+---------------+-----------------+-------------+-------------------------------+
--------------+-----------+---------------+----------------+----------------+-----------------+---------------+------------
 3289 |    16385 | replication | postgresql-192.168.2.247 | 192.168.2.247 |                 |       34902 | 2017-02-02 17:49:33.556689+11 |
              | streaming | 0/6000220     | 0/6000220      | 0/6000220      | 0/6000220       |             1 | sync
 3290 |    16385 | replication | postgresql-192.168.2.245 | 192.168.2.245 |                 |       43161 | 2017-02-02 17:49:34.761727+11 |
              | streaming | 0/6000220     | 0/6000220      | 0/6000220      | 0/6000220       |             1 | potential
 3296 |    16385 | replication | postgresql-192.168.2.246 | 192.168.2.246 |                 |       34566 | 2017-02-02 17:49:39.584053+11 |
              | streaming | 0/6000220     | 0/6000220      | 0/6000220      | 0/6000220       |             1 | potential
 3297 |    16385 | replication | postgresql-192.168.2.248 | 192.168.2.248 |                 |       38643 | 2017-02-02 17:49:41.835953+11 |
              | streaming | 0/6000220     | 0/6000220      | 0/6000220      | 0/6000220       |             1 | potential
(4 rows)

postgres=# show synchronous_standby_names ;
 synchronous_standby_names
---------------------------
 *
(1 row)
```

**Comment by CyberDem0n** [comment]
Hello,

```json
  "patroni": {
    "scope": "postgresql",
    "version": "1.1"
  },
```

The problem is that you are using old version of Patroni.
"synchronous_mode" is available starting from 1.2.

Please upgrade to 1.2.3 and it will work.

**Comment by jamessewell** [comment]
Thanks!
